### PR TITLE
Object Detection , Segmentation and Pose Model Loading

### PR DESCRIPTION
**Pull request by Akash-guna**
Adds Object Detection , Segmentation and Pose Detection Model Loading

Previous PR #6

### CODE REVIEW

File: core/utils/modelutils.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+
+def find_state_or_model(loaded):
+    """
+    Identifies if loaded model is a state dict or a nn.module
+    """
+    if isinstance(loaded, OrderedDict) or isinstance(loaded, dict):
+        return "STATE_DICT"
+    else:
+        return "MODEL"
```
Comment (shwu-nyunai) [inline core/utils/modelutils.py:129]: do we need this method to infer this? we can directly perform an isinstance check and derive if a loaded weight is model or state_dict.

if we are keeping this logic, we should also confirm the type for the model
```suggestion
def get_state_dict_or_model(loaded):
    """
    Identifies if loaded file is a state dict or a nn.module
    """
    if isinstance(loaded, OrderedDict, dict):
        return "STATE_DICT"
    elif isinstance(loaded, nn.Module): ## add imports
        return "MODEL"
```

File: core/utils/modelutils.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+
+def load_model_or_weights(path):
+    "Loads the custom model / weights from the provided path"
+    loaded = torch.load(path)
+    if find_state_or_model(loaded) == "MODEL":
+        return loaded, 1
+    else:
+        return loaded, 0
```
Comment (shwu-nyunai) [inline core/utils/modelutils.py:138]: as mentioned above, the two methods can be clubbed into one - 

```suggestion
def load_model_or_weights(path):
    "Loads the custom model / weights from the provided path"
    load = torch.load(path)
    if isinstance(load, nn.Module): ## add imports as necessary
        return load, 1
    elif isinstance(loaded, OrderedDict, dict):
        return load, 0
```

but imo, we can rather have a method that accepts a model and load_path and set the model accordingly.
would like to know more on the use case of this method.

**Review by shwu-nyunai — CHANGES_REQUESTED** [review]

File: core/models/ultralytics.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    return model
+
+
+def get_RTDETR(name, pretrained, cache_dir):
+    present_dir = os.getcwd()
+    os.chdir(cache_dir)
+    model = RTDETR(f"{name}.pt")
+    print(model)
+    os.chdir(present_dir)
+    return model
```
[the remark this answers is not in the document]
Comment (Akash-guna) [reply core/models/ultralytics.py]: Actually Ultralytics is currently not used for any models so it had to be removed added unneededly

File: core/utils/modelutils.py (hunk 1 above)
Comment (Akash-guna) [reply core/utils/modelutils.py]: made this change

**Review by shwu-nyunai — APPROVED** [review]
lgtm.

**Pull request merged**